helmnet/spectral.py

```diff
@@ -29,26 +29,45 @@ def conj(x):
         ix = x[..., 1]
     return torch.stack([rx, -ix], dim=-1)
 
-
 @torch.jit.script
 def fast_laplacian_with_pml(u, kx, ky, kx_sq, ky_sq, ax, bx, ay, by):
     """
     ax,bx are the 1/gamma and gamma'/gamma^3 coefficients in the laplacian operator of the paper, for the x axis
     """
     # TODO: redo this function before 9pm
     # Make 2d fourier transform of signal
-    u_fft = torch.fft(u, signal_ndim=2, normalized=False)
+
+    #u_fft = torch.fft(u, signal_ndim=2, normalized=False)
+    #since 1.6 the fft is a module with different API
+    u_fft = torch.view_as_real(
+        torch.fft.fftn(
+            torch.view_as_complex(u), 
+            dim = (-2, -1), 
+            norm = "backward"
+        )
+    )
 
     # get derivatives
-    dx = complex_mul(u_fft, kx)
-    dy = complex_mul(u_fft, ky)
+    dx  = complex_mul(u_fft, kx)
+    dy  = complex_mul(u_fft, ky)
```

Comment:
Those `complex_mul` and `conj` functions were used because pytorch didn't support backpropagation trough operations with complex numbers.
Do we still need them, or can pytorch now work with complex values and we can replace them with built-in functions?

Reply:
According to docs, pytorch autograd now works with complex operations, however JIT subsystem is still not supported. https://pytorch.org/docs/stable/complex_numbers.html